src/models/util/ParseCalChart3.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * CalChart3 was a bytestream with a series of 4 char codes describing the next
+ * section.  In general, the layout is:
+ *  <4char> <size_of_data> <data...> <"END "> <4char>
+ * with the exception of the first "INGL" block and the version the Version
+ * block is either "GURK", for CalChart3.0, or "GUXY" where X and Y are version
+ * numbers in hex.
+ *
+ * For CalChart 3.3 and later there are consistent 4char blocks that contain
+ * more elements of the show.
+ */
+
+import Show from '../Show';
+import StuntSheet from '../StuntSheet';
+import StuntSheetDot from '../StuntSheetDot';
+import { ReadInt32,
+  ReadInt16,
+  ReadStringTillEnd,
+  ReadArrayOfStringsTillEnd,
+  SplitDataViewIntoChunks,
+  CalChart3To4XConvert,
+  CalChart3To4YConvert } from './ParseCalChartUtils';
+import { ParseCalChart } from './ParseCalChart';
+
+// calchart 3.4 is a series of tables, with each one have the format: name_32,
+// size_32, data, end_32
+export class ParseCalChart3 implements ParseCalChart {
+  private numberDots: number = 0;
+
+  ParseShow(inputBuffer: ArrayBuffer): Show {
+    // we know the header for a CalChart3.5 show is 8 bytes.  Remove it.
+    const buffer = new DataView(inputBuffer, 8, inputBuffer.byteLength-8);
+    const split = SplitDataViewIntoChunks(buffer);
+    if (split.length !== 1 || split[0][0] !== 'SHOW') {
+      throw 'Cannot find show when parsing CalChart3 file';
+    }
+    return this.ParseCalChart3SHOW(split[0][1]);
+  }
+
+  ParseCalChart3SHOW(block: DataView): Show {
+    const show = new Show({
+      title: '',
+      stuntSheets: [],
+    });
+    const split = SplitDataViewIntoChunks(block);
+
+    for (const block of split) {
+      if (block[0] === 'SIZE') {
```

Comment:
Let's do a `switch` instead of four `if`s

Reply:
Done in next update!